packages/jellyfish-wallet-encrypted/__tests__/scryptsy.test.ts

```diff
@@ -0,0 +1,38 @@
+import { SimpleScryptsy } from '../src'
+
+describe('passphraseToKey()', () => {
+  it('Should be able to generate from random (same) passphrase into multiple desired length secret', () => {
+    const scryptsy = new SimpleScryptsy()
+    const secret1 = scryptsy.passphraseToKey('random password', Buffer.from('abcd', 'hex'), 64)
+    expect(secret1.length).toStrictEqual(64)
+
+    const secret2 = scryptsy.passphraseToKey('random password', Buffer.from('abcd', 'hex'), 40)
+    expect(secret2.length).toStrictEqual(40)
+  })
+
+  it('Configurable params (difficulty)', () => {
+    const easy = new SimpleScryptsy({
+      N: 16384, // 2 ^ 14
+      r: 8,
+      p: 1
+    })
+
+    const hard = new SimpleScryptsy({
+      N: 16384, // 2 ^ 14
+      r: 8,
+      p: 8
+    })
+
+    const easyStart = Date.now()
+    easy.passphraseToKey('random password', Buffer.from('abcd', 'hex'), 32)
+    const easyTime = Date.now() - easyStart
+
+    const hardStart = Date.now()
+    hard.passphraseToKey('random password', Buffer.from('abcd', 'hex'), 32)
+    const hardTime = Date.now() - hardStart
+
+    // significantly slower
+    // technically it is 8x harder, but they can be processed in parallel
+    expect(hardTime).toBeGreaterThan(easyTime * 2)
```

Comment:
what is the `ms` for these params?

Reply:
Using original suggested from bip0038, I think it takes 1-2 seconds to encrypt or decrypt

Reply:
I think can be 3-4 seconds depend on how powerful is the phone.